### PR TITLE
Change from GDI+ to Skia

### PR DESCRIPTION
**Pull request by fourst4r**
Because of the fact that GDI+ is notoriously slow (this is the backend to System.Drawing) I thought it would be nice to use Skia for graphics instead since its already a dependency.
In doing so I had to refactor quite a bit just replacing the System.Drawing stuff with SkiaSharp stuff, but after the work I noticed a 40-60% less CPU usage and 20-30% less memory! A great success, since it used to lag for me on fullscreen. Merge if u want

### CODE REVIEW

**Comment by Pr2FreeRunner** [comment]
Sounds ambitious, I will need to test this before I merge it. 👍

**Comment by Pr2FreeRunner** [comment]
Great job, I am impressed!

Interesting results, it feelz a lot smoother now, no "flickering" effect, which is great. 

Less lag in general when building (higher zoom levels)

It feelz like i am moving faster now compared to before, when pressing arrow keys and moving the camera view. Prob cuz faster (actual) FPS. 

I did sadly notice some major lag when using 5% zoom level tho, the arrow keys starts to lag af. Does 5% zoom level lag on ur computer as well?

Either way, great work, imma keep testing and analyze it. In the weekend I will merge it. 👍

**Comment by fourst4r** [comment]
I thought the FPS should be the same cuz I just added `ViewModel.Game.Engine.OnFrame += SKControl.InvalidateVisual;` which forces the control to paint, calling the paint handler, which calls `OnFrameUpdate`.

It lags for me also on 5% zoom level, I think this is unavoidable (unless GPU rendering?). There are more possibilities for optimization though via `DrawAtlas` to draw many bitmaps at once instead of doing the calls in a loop by `DrawBitmap`. 
Another thing is it may help to use the built-in transformation methods for offsets/zooming/rotation (SKMatrix). I don't fully understand how the zooming works yet.

**Comment by fourst4r** [comment]
Hmm it completely freezes at low zooms i guess because the frame is being drawn in the UI thread and its taking too long, every frame. Did not realize this before.
I will think about how to do the rendering in another thread...

File: BlockEditor/Models/Game/GameEngine.cs [hunk 1]
```diff
@@ -34,13 +35,13 @@ public bool Pause
...
         {
             Pause = false;
-            Thread.Sleep(GameEngine.FPS * 5); // wait for engine to pause
+            Thread.Sleep((int)(MsPerFrame * 5)); // wait for engine to pause
         }
 
         public GameEngine()
         {
             _timer = new System.Timers.Timer();
-            _timer.Interval = FPS;
```
Comment (fourst4r) [inline BlockEditor/Models/Game/GameEngine.cs:43]: This was making it update at 37fps

**Comment by Pr2FreeRunner** [comment]
The first version of the tool was actually multi threaded, but the problem was "flickering" effect. So i remade everything to a single thread. Which solved the flickering problems. Ofc the performance went down with it but it was good enough.

File: BlockEditor/Models/Game/GameEngine.cs (hunk 1 above)
Comment (Pr2FreeRunner) [reply BlockEditor/Models/Game/GameEngine.cs:43]: lel, i see now the 37ms FPS bug.

I was using 27ms between each frame, which gives an FPS of 37ms xD

Good job on fixing. 

Maybe we can use a slower FPS on zoom level 5%, so it lags less?
I also have another idea on fixing the lag on zoom level 5%...hmm..imma rejoin on discord so its easier to talk :p

**Pull request merged**